### PR TITLE
[Docs] `mount`/`shallow`: `update`: improve example

### PR DESCRIPTION
**Pull request by cadence09**
made a clarification on using update() method.

### CODE REVIEW

File: docs/api/ReactWrapper/update.md [hunk 1]
```diff
@@ -17,21 +17,28 @@ NOTE: this does not force a re-render. Use `wrapper.setProps({})` to force a re-
...
 
 ```jsx
-class ImpureRender extends React.Component {
+class UpdateEnzyme extends React.Component {
   constructor(props) {
     super(props);
-    this.count = 0;
+    this.state={
+      count:0
+    }
```
Comment (ljharb) [inline docs/api/ReactWrapper/update.md:25]:
```suggestion
    this.state = {
      count: 0
    };
    this.increment = this.increment.bind(this);
```

File: docs/api/ReactWrapper/update.md [hunk 2]
```diff
@@ -17,21 +17,28 @@ NOTE: this does not force a re-render. Use `wrapper.setProps({})` to force a re-
...
+  }
+  increment=()=>{
+    this.setState({count:this.state.count+1})
   }
```
Comment (ljharb) [inline docs/api/ReactWrapper/update.md:29]:
```suggestion

  increment() {
    this.setState({ count: this.state.count + 1 })
  }
  
```

File: docs/api/ReactWrapper/update.md [hunk 3]
```diff
@@ -17,21 +17,28 @@ NOTE: this does not force a re-render. Use `wrapper.setProps({})` to force a re-
...
-
   render() {
-    this.count += 1;
-    return <div>{this.count}</div>;
+    return <button className='increment' onClick={this.increment}>{this.state.count}</button>;
```
Comment (ljharb) [inline docs/api/ReactWrapper/update.md:31]:
```suggestion
    const { count } = this.state;
    return <button className="increment" onClick={this.increment}>{count}</button>;
```

File: docs/api/ReactWrapper/update.md [hunk 4]
```diff
@@ -17,21 +17,28 @@ NOTE: this does not force a re-render. Use `wrapper.setProps({})` to force a re-
...
-wrapper.update();
-expect(wrapper.text()).to.equal('1');
+test('increment by 1', ()=>{
+  const wrapper = shallow(<UpdateEnzyme />);
+  expect(wrapper.find('button.increment').text()).toEqual('0');
+  wrapper.instance().increment()
+  // Update Enzyme's view of output
+  wrapper.update();
+  expect(wrapper.find('button.increment').text()).toEqual('1');
+})
```
Comment (ljharb) [inline docs/api/ReactWrapper/update.md:43]: enzyme's docs use mocha and chai syntax, not jest syntax.
```suggestion
const wrapper = shallow(<UpdateEnzyme />);
expect(wrapper.find('button.increment').text()).to.equal('0');
wrapper.instance().increment();
// Update Enzyme's view of output
wrapper.update();
expect(wrapper.find('button.increment').text()).to.equal('1');
```

**Comment by cadence09** [comment]
Yes, I already did committed the changes.

**Comment by ljharb** [comment]
@cadence09 thanks! could you make the same changes in https://github.com/enzymejs/enzyme/blob/master/docs/api/ShallowWrapper/update.md ?

**Comment by cadence09** [comment]
This time I made changes in the ShallowWrapper folder.

**Pull request merged**